src/ir/LocalGraph.cpp

```diff
@@ -421,4 +521,10 @@ void LocalGraph::computeSSAIndexes() {
 
 bool LocalGraph::isSSA(Index x) { return SSAIndexes.count(x); }
 
+void LocalGraph::computeGetSets(LocalGet* get) const {
+  assert(mode == Mode::Lazy);
```

Comment:
Since the lazy and eager modes expose different methods, how about making them separate subclasses of a shared base class?

Reply:
I experimented with this in the last commits. It is a little unpleasant since we want to do work in the constructor, which is different in the two cases. So the subclass must tell the parent not to do its work somehow, or use a virtual method, or have a "post-constructor init" function, or such. I'm not sure how much I like those details, but I agree it makes the API a little nicer to use `LazyLocalGraph`, and also it makes the separation cleaner. It also may be faster, since we aren't doing an `if` on the mode in each call to `getSets`, so I lean towards the current state with that change, with reservations, PTAL.

Reply:
Another option might be to have three classes, LocalGraphBase (no computation), LocalGraph (eager), LazyLocalGraph.

Reply:
That's what I originally had in mind. It might be simpler, but the current code LGTM as well.

Reply:
Yeah, a base class does feel better the more I think about it. I implemented that now, sorry for all the iterations here, but PTAL, now I think this is in good shape.